packages/store/src/legacyMap.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2019 Agoric, under Apache license 2.0
+
+import { assert, details as X, q } from '@agoric/assert';
+
+/**
+ * @deprecated switch to ScalarMap if possible, Map otherwise
+ * @param {string} [keyName='key'] - the column name for the key
+ */
+export const makeLegacyMap = (keyName = 'key') => {
+  const m = new Map();
+  const assertKeyDoesNotExist = key =>
+    assert(!m.has(key), X`${q(keyName)} already registered: ${key}`);
+  const assertKeyExists = key =>
+    assert(m.has(key), X`${q(keyName)} not found: ${key}`);
+  const legacyMap = {
+    has: key => {
+      // .has is very accepting
```

Comment:
What does this mean?

Reply:
`has` comment changed to

```js
      // Check if a key exists. The key can be any JavaScript value,
      // though the answer will always be false for keys that cannot be found
      // in this map.
```

This is actually the was JS Map and WeakMap work as well. You can ask a JS WeakMap if it `has(8)`. The answer is always `false`, not `TypeError`. Resolving.